Tests/MeshLib/TestCoordinatesMappingLocal.cpp

```diff
@@ -0,0 +1,270 @@
+    static const unsigned e_nnodes = ElementType::n_all_nodes;
+
+    static MeshLib::Line* createY()
+    {
+        MeshLib::Node** nodes = new MeshLib::Node*[e_nnodes];
+        nodes[0] = new MeshLib::Node(0.0, -1.0, 0.0);
+        nodes[1] = new MeshLib::Node(0.0,  1.0, 0.0);
+        return new MeshLib::Line(nodes);
+    }
+
+    static MeshLib::Line* createZ()
+    {
+        MeshLib::Node** nodes = new MeshLib::Node*[e_nnodes];
+        nodes[0] = new MeshLib::Node(0.0, 0.0, -1.0);
+        nodes[1] = new MeshLib::Node(0.0, 0.0,  1.0);
+        return new MeshLib::Line(nodes);
+    }
+
+    static MeshLib::Line* createXY()
+    {
+        // 45degree inclined
+        MeshLib::Node** nodes = new MeshLib::Node*[e_nnodes];
+        nodes[0] = new MeshLib::Node(0.0, 0.0, 0.0);
+        nodes[1] = new MeshLib::Node(2./sqrt(2), 2./sqrt(2), 0.0);
+        return new MeshLib::Line(nodes);
+    }
+
+    static MeshLib::Line* createXYZ()
+    {
+        MeshLib::Node** nodes = new MeshLib::Node*[e_nnodes];
+        nodes[0] = new MeshLib::Node(0.0, 0.0, 0.0);
+        nodes[1] = new MeshLib::Node(2./sqrt(3), 2./sqrt(3), 2./sqrt(3));
+        return new MeshLib::Line(nodes);
+    }
+
+};
+
+class TestQuad4
+{
+ public:
+    // Element information
+    typedef MeshLib::Quad ElementType;
+    static const unsigned dim = 2; //ElementType::dimension;
+    static const unsigned e_nnodes = ElementType::n_all_nodes;
+
+    // 2.5D case: inclined
+    static MeshLib::Quad* createXYZ()
+    {
+        MeshLib::Node** nodes = new MeshLib::Node*[e_nnodes];
+        // rotate 40 degree around x axis
+        nodes[0] = new MeshLib::Node( 1.0,  0.7071067811865475,  0.7071067811865475);
+        nodes[1] = new MeshLib::Node(-1.0,  0.7071067811865475,  0.7071067811865475);
+        nodes[2] = new MeshLib::Node(-1.0, -0.7071067811865475, -0.7071067811865475);
+        nodes[3] = new MeshLib::Node( 1.0, -0.7071067811865475, -0.7071067811865475);
+        return new MeshLib::Quad(nodes);
+    }
+
+    // 2.5D case: inclined
+    static MeshLib::Quad* createXZ()
+    {
+        MeshLib::Node** nodes = new MeshLib::Node*[e_nnodes];
+        nodes[0] = new MeshLib::Node( 1.0, 0.0,  1.0);
+        nodes[1] = new MeshLib::Node(-1.0, 0.0,  1.0);
+        nodes[2] = new MeshLib::Node(-1.0, 0.0, -1.0);
+        nodes[3] = new MeshLib::Node( 1.0, 0.0, -1.0);
+        return new MeshLib::Quad(nodes);
+    }
+
+    // 2.5D case: inclined
+    static MeshLib::Quad* createYZ()
+    {
+        MeshLib::Node** nodes = new MeshLib::Node*[e_nnodes];
+        nodes[0] = new MeshLib::Node(0.0,  1.0,  1.0);
+        nodes[1] = new MeshLib::Node(0.0, -1.0,  1.0);
+        nodes[2] = new MeshLib::Node(0.0, -1.0, -1.0);
+        nodes[3] = new MeshLib::Node(0.0,  1.0, -1.0);
+        return new MeshLib::Quad(nodes);
+    }
+
+};
+
+void debugOutput(MeshLib::Element *ele, MeshLib::ElementCoordinatesMappingLocal &mapping)
+{
+    std::cout.precision(12);
```

Comment:
Why not logog? Do you plan to put this code between the preprocessor directives `#ifndef NDEBUG` and `#endif`?